### PR TITLE
test(JestMigration): PieSpec.js -> Pie.spec.tsx

### PR DESCRIPTION
**Pull request by james2hey**
## Description

Migrate Pie.js -> Pie.spec.tsx

## Related Issue
#3227

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test-jest/polar/Pie.spec.tsx [hunk 1]
```diff
@@ -25,19 +21,27 @@ describe('<Pie />', () => {
...
+          innerRadius={0}
+          outerRadius={200}
+          sectors={sectors}
+          dataKey="cy"
+        />
       </Surface>,
     );
 
-    expect(wrapper.find('.recharts-pie-sector').length).to.equal(sectors.length);
+    expect(container.querySelectorAll('.recharts-pie-sector').length).toBe(sectors.length);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:39]:
```suggestion
    expect(container.querySelectorAll('.recharts-pie-sector')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 2]
```diff
@@ -48,16 +52,17 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-active-shape').length).to.equal(1);
+    expect(container.querySelectorAll('.customized-active-shape').length).toBe(1);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:60]:
```suggestion
    expect(container.querySelectorAll('.customized-active-shape')).toHaveLength(1);
```

File: test-jest/polar/Pie.spec.tsx [hunk 3]
```diff
@@ -68,15 +73,16 @@ describe('<Pie />', () => {
...
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:81]:
```suggestion
    expect(container.querySelectorAll('.customized-active-shape')).toHaveLength(1);
```

File: test-jest/polar/Pie.spec.tsx [hunk 4]
```diff
@@ -87,18 +93,19 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-active-shape').length).to.equal(0);
+    expect(container.querySelectorAll('.customized-active-shape').length).toBe(0);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:101]:
```suggestion
    expect(container.querySelectorAll('.customized-active-shape')).toHaveLength(0);
```

File: test-jest/polar/Pie.spec.tsx [hunk 5]
```diff
@@ -110,16 +117,19 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
-    expect(wrapper.find('.customized-inactive-shape').length).to.equal(4);
+    expect(container.querySelectorAll('.customized-inactive-shape').length).toBe(4);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:124]:
```suggestion
    expect(container.querySelectorAll('.customized-inactive-shape')).toHaveLength(4);
```

File: test-jest/polar/Pie.spec.tsx [hunk 6]
```diff
@@ -131,14 +141,15 @@ describe('<Pie />', () => {
...
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:148]:
```suggestion
    expect(container.querySelectorAll('.customized-inactive-shape')).toHaveLength(4);
```

File: test-jest/polar/Pie.spec.tsx [hunk 7]
```diff
@@ -150,17 +161,20 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-inactive-shape').length).to.equal(0);
+    expect(container.querySelectorAll('.customized-inactive-shape').length).toBe(0);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:169]:
```suggestion
    expect(container.querySelectorAll('.customized-inactive-shape')).toHaveLength(0);
```

File: test-jest/polar/Pie.spec.tsx [hunk 8]
```diff
@@ -171,15 +185,16 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
-    expect(wrapper.find('.customized-inactive-shape').length).to.equal(0);
+    expect(container.querySelectorAll('.customized-inactive-shape').length).toBe(0);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:192]:
```suggestion
    expect(container.querySelectorAll('.customized-inactive-shape')).toHaveLength(0);
```

File: test-jest/polar/Pie.spec.tsx [hunk 9]
```diff
@@ -190,23 +205,24 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-active-shape').length).to.equal(2);
+    expect(container.querySelectorAll('.customized-active-shape').length).toBe(2);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:213]:
```suggestion
    expect(container.querySelectorAll('.customized-active-shape')).toHaveLength(2);
```

File: test-jest/polar/Pie.spec.tsx [hunk 10]
```diff
@@ -216,19 +232,20 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-label').length).to.equal(sectors.length);
+    expect(container.querySelectorAll('.customized-label').length).toBe(sectors.length);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:240]:
```suggestion
    expect(container.querySelectorAll('.customized-label')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 11]
```diff
@@ -238,26 +255,27 @@ describe('<Pie />', () => {
...
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
     setTimeout(() => {
-      expect(wrapper.find('.recharts-pie-label-text').length).to.equal(sectors.length);
-      expect(wrapper.find('.recharts-pie-label-text').first().text()).to.equal('A: 40');
+      expect(container.querySelectorAll('.recharts-pie-label-text').length).toBe(sectors.length);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:264]:
```suggestion
      expect(container.querySelectorAll('.recharts-pie-label-text')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 12]
```diff
@@ -267,22 +285,23 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-label').length).to.equal(sectors.length);
+    expect(container.querySelectorAll('.customized-label').length).toBe(sectors.length);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:293]:
```suggestion
    expect(container.querySelectorAll('.customized-label')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 13]
```diff
@@ -293,22 +312,23 @@ describe('<Pie />', () => {
           innerRadius={0}
           outerRadius={200}
           sectors={sectors}
+          dataKey="cy"
         />
       </Surface>,
     );
 
-    expect(wrapper.find('.customized-label-line').length).to.equal(sectors.length);
+    expect(container.querySelectorAll('.customized-label-line').length).toBe(sectors.length);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:320]:
```suggestion
    expect(container.querySelectorAll('.customized-label-line')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 14]
```diff
@@ -319,28 +339,30 @@ describe('<Pie />', () => {
...
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:347]:
```suggestion
    expect(container.querySelectorAll('.customized-label-line')).toHaveLength(sectors.length);
```

File: test-jest/polar/Pie.spec.tsx [hunk 15]
```diff
@@ -319,28 +339,30 @@ describe('<Pie />', () => {
...
+  test("Don't render any sector when data is empty", () => {
+    const { container } = render(
       <Surface width={500} height={500}>
-        <Pie sectors={[]} />
+        <Pie sectors={[]} dataKey="cy" />
       </Surface>,
     );
 
-    expect(wrapper.find('.recharts-pie').length).to.equal(0);
+    expect(container.querySelectorAll('.recharts-pie').length).toBe(0);
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:357]:
```suggestion
    expect(container.querySelectorAll('.recharts-pie')).toHaveLength(0);
```

File: test-jest/polar/Pie.spec.tsx [hunk 16]
```diff
@@ -352,83 +374,110 @@ describe('<Pie />', () => {
...
-    const se = wrapper.find(Layer).at(3);
-
-    se.simulate('mouseEnter');
-    expect(onMouseEnter.calledOnce).to.equal(true);
-    se.simulate('mouseLeave');
-    expect(onMouseLeave.calledOnce).to.equal(true);
-    se.simulate('click');
-    expect(onClick.calledOnce).to.equal(true);
+
+    const se = container.querySelectorAll('.recharts-layer')[4];
```
Comment (proke03) [inline test-jest/polar/Pie.spec.tsx:382]: Please do not use abbreviations.
Did you mean the sector?

**Review by proke03 — COMMENTED** [review]
All suggestions are to change the method from toBe to toHaveLength.

File: test-jest/polar/Pie.spec.tsx [hunk 17]
```diff
@@ -25,19 +21,27 @@ describe('<Pie />', () => {
...
 
-    expect(wrapper.find('.recharts-pie-sector').length).to.equal(sectors.length);
+    expect(container.querySelectorAll('.recharts-pie-sector').length).toBe(sectors.length);
   });
 
-  it('Render customized active sector when activeShape is set to be an element', () => {
-    const ActiveShape = props => <Sector {...props} fill="#ff7300" className="customized-active-shape" />;
-    const wrapper = render(
+  test('Render customized active sector when activeShape is set to be an element', () => {
+    const ActiveShape = (props: any) => <Sector {...props} fill="#ff7300" className="customized-active-shape" />;
```
Comment (ckifer) [inline test-jest/polar/Pie.spec.tsx:43]: please avoid any where possible/if possible (might not be easy to here, but just a general comment)

File: test-jest/polar/Pie.spec.tsx (hunk 16 above)
Comment (ckifer) [reply test-jest/polar/Pie.spec.tsx:382]: +1 not sure what `se` is - this was in the previous test but lets expand it out

File: test-jest/polar/Pie.spec.tsx [hunk 18]
```diff
@@ -352,83 +374,110 @@ describe('<Pie />', () => {
...
+            dataKey="cy"
+          />
         </Surface>
       </div>,
     );
     const pie = container.getElementsByClassName('recharts-pie')[0];
     const pieContainer = document.getElementsByClassName('container')[0];
 
-    pieContainer.focus();
+    (pieContainer as any).focus();
```
Comment (ckifer) [inline test-jest/polar/Pie.spec.tsx:420]: any way this doesn't have to be `any`?

**Review by ckifer — CHANGES_REQUESTED** [review]
Agreed with the other suggested changes. Added a few additional comments on types

File: test-jest/polar/Pie.spec.tsx (hunk 17 above)
Comment (james2hey) [reply test-jest/polar/Pie.spec.tsx:47]: I added some additional types for the `label` and `labelLine` so we don't have to use `any`.

But the `activeShape` and `inactiveShape` both have types `((props: any) => ReactElement<SVGElement>)` so I have left these with `any` to match this type.

File: test-jest/polar/Pie.spec.tsx (hunk 16 above)
Comment (james2hey) [reply test-jest/polar/Pie.spec.tsx]: Yeah it was carried over from the original file, I've expanded it out to `sector`.

File: test-jest/polar/Pie.spec.tsx (hunk 18 above)
Comment (james2hey) [reply test-jest/polar/Pie.spec.tsx]: I've casted the type to an `HTMLElement` to preserve typing.

File: test-jest/polar/Pie.spec.tsx (hunk 17 above)
Comment (ckifer) [reply test-jest/polar/Pie.spec.tsx:46]: Thanks!

**Review by ckifer — APPROVED** [review]
LGTM 🚀

**Pull request merged**